### PR TITLE
Adding arbitrary mapping feature in execution plan.

### PR DESCRIPTION
**Pull request by ksdperera**

### CODE REVIEW

File: components/event-processor/org.wso2.carbon.event.processor.core/src/main/java/org/wso2/carbon/event/processor/core/internal/util/helper/EventProcessorHelper.java [hunk 1]
```diff
@@ -208,14 +208,29 @@ private static boolean validateSiddhiStreamWithDatabridgeStream(String streamNam
...
                         + (streamDefinition.getPayloadData() == null ? 0 : streamDefinition.getPayloadData().size());
-                if (siddhiStreamDefinition.getAttributeList().size() != streamSize) {
+                Element ArbitraryElement = AnnotationHelper.getAnnotationElement(EventProcessorConstants.ANNOTATION_IMPORT,
+                        EventProcessorConstants.ANNOTATION_INCLUDE_ARBITRARY, siddhiStreamDefinition.getAnnotations());
+                if (ArbitraryElement == null) {
+                    ArbitraryElement = AnnotationHelper.getAnnotationElement(EventProcessorConstants.ANNOTATION_EXPORT,
+                            EventProcessorConstants.ANNOTATION_INCLUDE_ARBITRARY, siddhiStreamDefinition.getAnnotations());
+                }
+                if (ArbitraryElement != null && ArbitraryElement.getValue().equals("true")) {
+                    if (!siddhiStreamDefinition.getAttributeList().get(siddhiStreamAttributeCount - 1).getName().equals(EventProcessorConstants.ARBITRARY_MAP)) {
```
Comment (tishan89) [inline components/event-processor/org.wso2.carbon.event.processor.core/src/main/java/org/wso2/carbon/event/processor/core/internal/util/helper/EventProcessorHelper.java]: Exceeds recommended line length

File: components/event-processor/org.wso2.carbon.event.processor.ui/src/main/resources/web/eventprocessor/js/create_execution_plan_helper.js [hunk 2]
```diff
@@ -180,17 +181,33 @@ function addImportedStreamDefinition() {
...
-                SIDDHI_SINGLE_QUOTE + streamId + SIDDHI_SINGLE_QUOTE +
-                ANNOTATION_TOKEN_CLOSING_BRACKET +
-                SIDDHI_LINE_BREAK +
-                SIDDHI_LITERAL_DEFINE_STREAM + SIDDHI_SPACE_LITERAL + streamAs + SIDDHI_SPACE_LITERAL +
-                ANNOTATION_TOKEN_OPENING_BRACKET +
-                streamDefinition +
-                ANNOTATION_TOKEN_CLOSING_BRACKET +
-                SIDDHI_STATEMENT_DELIMETER;
+            var importStatement;
+            if(propImportArbitrary){
```
Comment (thiliA) [inline components/event-processor/org.wso2.carbon.event.processor.ui/src/main/resources/web/eventprocessor/js/create_execution_plan_helper.js]: @ksdperera Let's format this code block

File: components/event-processor/org.wso2.carbon.event.processor.ui/src/main/resources/web/eventprocessor/js/create_execution_plan_helper.js [hunk 3]
```diff
@@ -180,17 +181,33 @@ function addImportedStreamDefinition() {
...
+                    SIDDHI_SINGLE_QUOTE + streamId + SIDDHI_SINGLE_QUOTE +
+                    INCLUDE_ARBITRARY_DATA +
+                    ANNOTATION_TOKEN_CLOSING_BRACKET +
+                    SIDDHI_LINE_BREAK +
+                    SIDDHI_LITERAL_DEFINE_STREAM + SIDDHI_SPACE_LITERAL + streamAs + SIDDHI_SPACE_LITERAL +
+                    ANNOTATION_TOKEN_OPENING_BRACKET +
+                    streamDefinition +
+                    ANNOTATION_TOKEN_CLOSING_BRACKET +
+                    SIDDHI_STATEMENT_DELIMETER;
+            }else{
```
Comment (thiliA) [inline components/event-processor/org.wso2.carbon.event.processor.ui/src/main/resources/web/eventprocessor/js/create_execution_plan_helper.js]: @ksdperera Code formatting needed

File: components/event-processor/org.wso2.carbon.event.processor.core/src/main/java/org/wso2/carbon/event/processor/core/internal/util/helper/EventProcessorHelper.java (hunk 1 above)
Comment (ksdperera) [inline components/event-processor/org.wso2.carbon.event.processor.core/src/main/java/org/wso2/carbon/event/processor/core/internal/util/helper/EventProcessorHelper.java]: Hi Tishan,

This is fixed by the commits in : https://github.com/wso2/carbon-event-processing/pull/266

Thanks,
SajithD

File: components/event-processor/org.wso2.carbon.event.processor.ui/src/main/resources/web/eventprocessor/js/create_execution_plan_helper.js (hunk 2 above)
Comment (ksdperera) [inline components/event-processor/org.wso2.carbon.event.processor.ui/src/main/resources/web/eventprocessor/js/create_execution_plan_helper.js]: Hi Thilini,

This is fixed by the commits in : https://github.com/wso2/carbon-event-processing/pull/266

Thanks,
SajithD

File: components/event-processor/org.wso2.carbon.event.processor.ui/src/main/resources/web/eventprocessor/js/create_execution_plan_helper.js (hunk 3 above)
Comment (ksdperera) [inline components/event-processor/org.wso2.carbon.event.processor.ui/src/main/resources/web/eventprocessor/js/create_execution_plan_helper.js]: Hi Thilini,

This is fixed by the commits in : https://github.com/wso2/carbon-event-processing/pull/266

Thanks,
SajithD

**Pull request merged**